### ISSUE TITLE
Brush size caps at 40 when changing it via code but does not cap the same way when trying to set it via  ipython console

### ISSUE THREAD

**Issue by pranathivemuri** [napari/napari#2745]
## 🐛 Bug
I have the below code in one of my programs when I run that the brush size is printed out as 40 which makes sense as https://github.com/napari/napari/blob/master/napari/_qt/layer_controls/qt_labels_controls.py#L415
but I would want the brush size to go over 40 pixels, so I tried to change it once the window opened with images. So, I set the brush_size with the same command in the ipython console generated using ``viewer.layers["binary"].brush_size =100`` it sets to 100. So, I am not sure why it isn't clipping once but it is clipping the other time, maybe the event is being generated vs not.

```
        viewer.layers["binary"].mode = "paint"
        viewer.layers["binary"].brush_size = int(100.0)
        viewer.layers["binary"].brush_shape = "circle"
        print("updated binary brush size")
````

## Expected behavior

1. I wanted to see if brush_size range could be set by the user instead of clipping it at 40 
2. I wanted to also find out why the events are inconsistently generated

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

 - Any other relevant information:

**Comment by sofroniewn** [comment]
Ah yeah this relates to how we're currently doing our sliders and if an event triggers the slider which then does the clip. We might want to think of updating the max allowed value of the slider rather than clipping. Curious what others think?

**Comment by pranathivemuri** [comment]
Wondering if anyone else has any input on this?

**Comment by sofroniewn** [comment]
@pranathivemuri there is some work going on in #2709 and #2753 that will allow more flexibility over our sliders. Once we've got those in we should be able to make this change more easily

**Comment by pranathivemuri** [comment]
thanks! :)

**Comment by sofroniewn** [comment]
@pranathivemuri if you comment out the clipping and instead update the max allowed value do things now work for you? We could just merge that without waiting on #2753 to be done 

https://github.com/napari/napari/blob/ed6b8ef2e55471b42a3ea47875b7e1ef0f99ba80/napari/_qt/layer_controls/qt_labels_controls.py#L434

**Comment by sofroniewn** [comment]
I have now implemented a fix for this in #2901
